polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -641,6 +667,7 @@ async fn validate_candidate_exhaustive(
 	pov: Arc<PoV>,
 	executor_params: ExecutorParams,
 	exec_kind: PvfExecKind,
+	request_ttl: Option<BlockNumber>,
```

Comment:
Instead of adding the Option, we can modify the `PvfExecKind` (not the runtime one) variants to include the ttl value.